packages/nextjs/src/config/wrappers/wrapperUtils.ts

```diff
@@ -1,25 +1,46 @@
+import { getActiveTransaction } from '@sentry/tracing';
+
 import { DataFetchingFunction } from './types';
 
 /**
- * Pass-through wrapper for the original function, used as a first step in eventually wrapping the data-fetching
- * functions with code for tracing.
+ * Create a span to track the wrapped function and update transaction name with parameterized route.
  *
  * @template T Types for `getInitialProps`, `getStaticProps`, and `getServerSideProps`
  * @param origFunction The user's exported `getInitialProps`, `getStaticProps`, or `getServerSideProps` function
  * @param context The context object passed by nextjs to the function
  * @returns The result of calling the user's function
  */
-export async function callOriginal<T extends DataFetchingFunction>(
-  origFunction: T['fn'],
-  context: T['context'],
-): Promise<T['result']> {
-  let props;
-
-  // TODO: Can't figure out how to tell TS that the types are correlated - that a `GSPropsFunction` will only get passed
-  // `GSPropsContext` and never, say, `GSSPContext`. That's what wrapping everything in objects and using the generic
-  // and pulling the types from the generic rather than specifying them directly was supposed to do, but... no luck.
-  // eslint-disable-next-line prefer-const, @typescript-eslint/no-explicit-any
-  props = await (origFunction as any)(context);
-
-  return props;
+export async function wrapperCore<T extends DataFetchingFunction>(options: {
+  origFunction: T['fn'];
+  context: T['context'];
+  route: string;
+  op: string;
+}): Promise<T['result']> {
+  const { origFunction, context, route, op } = options;
+
+  const transaction = getActiveTransaction();
+
+  if (transaction) {
+    // TODO: Make sure that the given route matches the name of the active transaction (to prevent background data
+    // fetching from switching the name to a completely other route)
```

Comment:
If we do totally ditch `instrumentServer`, assuming you can't combine `getInitialProps` with either of the other data fetchers (which we should test), then it's simpler than that, even, I think. Data fetchers will become like route handlers in that any given request hits only one, which means each one should be in its own domain/start its own transaction.

So yeah, maybe we don't even worry about the overwriting case for the moment, because it may well disappear. I'll update the PR description and comment accordingly.

Reply:
Sadly, when there is a `getInitialProps` in `_app.js` and you navigate to a page that has `getServerSideProps`, it runs GIP and GSSP right after. So we still might need to handle that case.

But yeah, otherwise this confirms my view on things. 👍